crates/snapshot/src/snapshotter.rs

```diff
@@ -109,10 +129,38 @@ impl<DB: Database> Snapshotter<DB> {
         }
     }
 
-    fn update_highest_snapshots_tracker(&self) {
+    /// Looks into the snapshot directory to find the highest snapshotted block of each segment, and
+    /// notifies every tracker.
+    fn update_highest_snapshots_tracker(&mut self) -> RethResult<()> {
+        // It walks over the directory and parses the snapshot filenames extracing `SnapshotSegment`
+        // and their inclusive range. It then takes the maximum block number for each specific
+        // segment.
+        for (segment, range) in std::fs::read_dir(&self.snapshots_path)
+            .map_err(|err| RethError::Custom(err.to_string()))?
+            .filter_map(Result::ok)
+            .filter_map(|entry| {
+                if let Ok(true) = entry.metadata().map(|metadata| metadata.is_file()) {
```

Comment:
```suggestion
                if entry.metadata().map_or(false, |metadata| metadata.is_file()) {
```

Reply:
or maybe even
```suggestion
                if entry.metadata().map(|metadata| metadata.is_file())? {
```
because the closure returns an option

Reply:
did it in another PR for the first option